homeassistant/components/saj/__init__.py

```diff
@@ -1 +1,49 @@
 """The saj component."""
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+
+from .const import DATA_INVERTER, DATA_UNDO_UPDATE_LISTENER, DOMAIN
+from .sensor import SAJInverter
+
+PLATFORMS = ["sensor"]
+
+
+async def async_setup(hass: HomeAssistant, config: dict):
+    """Set up the saj component."""
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Set up saj from a config entry."""
+
+    config = entry.data.copy()
+    config.update(entry.options)
+    inverter = SAJInverter(config)
```

Comment:
We should explicitly create the configuration and not just pass the dictionary as it. (which would remove the need for things like making a copy as well)

Reply:
I'm missing some kind of connection testing here?
The config flow does that, but on entry setup there is nothing?
Should it check if it has auth for example?

Reply:
I removed the connection check because these inverters were only available in the daytime.
Now, if you restart HA after sunset it became available the next day. 
If we had another method I'm open to that.

Reply:
When a connection fails, it can raise a ConfigEntryNotReady, HA will keep retrying.

Reply:
I will try this.

Reply:
It works, thanks